daemon/main/nzbget.cpp

```diff
@@ -272,11 +272,23 @@ void NZBGet::Init()
 	BootConfig();
 
 #ifndef WIN32
-	if (m_options->GetUMask() < 01000)
+	mode_t uMask = static_cast<mode_t>(m_options->GetUMask());
+	if (uMask > 0 && uMask < 01000)
 	{
 		/* set newly created file permissions */
-		umask(m_options->GetUMask());
+		FileSystem::uMask = uMask;
+		umask(FileSystem::uMask);
 	}
+	else
+	{
+		FileSystem::uMask = umask(022);
```

Comment:
```suggestion
		FileSystem::uMask = umask(0);
```

This should give you `022` in 99,99% of cases.